api/configuration/v1alpha1/key_set_ref.go

```diff
@@ -0,0 +1,47 @@
+package v1alpha1
+
+// KeySetRefType is the enum type for the KeySetRef.
+// +kubebuilder:validation:Enum=konnectID;namespacedRef
+type KeySetRefType string
+
+const (
+	// KeySetRefKonnectID is the type for the KonnectID KeySetRef.
+	// It is used to reference a KeySet entity by its ID on the Konnect platform.
+	KeySetRefKonnectID KeySetRefType = "konnectID"
+
+	// KeySetRefNamespacedRef is the type for the KeySetRef.
+	// It is used to reference a KeySet entity inside the cluster
+	// using a namespaced reference.
+	KeySetRefNamespacedRef KeySetRefType = "namespacedRef"
+)
+
+// KeySetRef is the schema for the KeySetRef type.
+// It is used to reference a KeySet entity.
+// +kubebuilder:validation:XValidation:rule="self.type == 'namespacedRef' ? has(self.namespacedRef) : true", message="when type is namespacedRef, namespacedRef must be set"
+// +kubebuilder:validation:XValidation:rule="self.type == 'konnectID' ? has(self.konnectID) : true", message="when type is konnectID, konnectID must be set"
+type KeySetRef struct {
+	// Type defines type of the KeySet object reference. It can be one of:
+	// - konnectID
+	// - namespacedRef
+	Type KeySetRefType `json:"type,omitempty"`
```

Comment:
Do we need the `omitempty` for anything here?

```suggestion
	Type KeySetRefType `json:"type"`
```